apps/protocols/x509/mode/validity.pm

```diff
@@ -156,6 +156,44 @@ sub run {
                                         short_msg => sprintf("Subject Name [%s] is present in Certificate", join(', ', @subject_matched)));
         }
 
+    #BOTH expiration and subject
+    } elsif ($self->{option_results}->{validity_mode} eq 'expiration-and-subject') {
+	my @subject_matched = ();
+
+        if ($subject =~ /$self->{option_results}->{subjectname}/mi) {
+            push @subject_matched, $subject;
+        } else {
+            $self->{output}->output_add(long_msg => sprintf("Subject Name '%s' is also present in Certificate", $subject), debug => 1);
+        }
+
+        my @subject_alt_names = Net::SSLeay::X509_get_subjectAltNames($cert);
+        for (my $i =  0; $i < $#subject_alt_names; $i += 2) {
+            my ($type, $name) = ($subject_alt_names[$i], $subject_alt_names[$i + 1]);
+            if ($type == GEN_IPADD) {
+                $name = inet_ntoa($name);
+            }
+            if ($name =~ /$self->{option_results}->{subjectname}/mi) {
+                push @subject_matched, $name;
+            } else {
+                $self->{output}->output_add(long_msg => sprintf("Subject Name '%s' is also present in Certificate", $name), debug => 1);
+            }
+        }
+
+        if (@subject_matched == 0) {
```

Comment:
If I understand this one, you check the subject and if it's not OK then you do not try to check expiration date right ?